terraform/azure/modules/1_nbs6/lbi/main.tf

```diff
@@ -31,12 +31,14 @@ resource "azurerm_lb" "lbi" {
 
 # Create Backend Pool
 resource "azurerm_lb_backend_address_pool" "lbi-pool" {
+  depends_on = [ azurerm_lb.lbi ]
   name            = "${var.prefix}-lbi-backend-pool"
```

Comment:
small change in the we are using the term "resource_prefix" instead of "prefix" in other modules. Likely requires updating variables.tf as well